libavcodec/qsvenc_h264.c

```diff
@@ -150,6 +150,13 @@ static const AVOption options[] = {
 
     { "repeat_pps", "repeat pps for every frame", OFFSET(qsv.repeat_pps), AV_OPT_TYPE_BOOL, { .i64 = 0 }, 0, 1, VE },
 
+#if QSV_HAVE_HE
+    { "dual_gfx", "Prefer processing on both iGfx and dGfx simultaneously", OFFSET(qsv.dual_gfx), AV_OPT_TYPE_INT, { .i64 = MFX_HYPERMODE_OFF }, MFX_HYPERMODE_OFF, MFX_HYPERMODE_ADAPTIVE, VE, "dual_gfx" },
+    { "off",       NULL, 0, AV_OPT_TYPE_CONST, { .i64 = MFX_HYPERMODE_OFF       },   INT_MIN, INT_MAX, VE, "dual_gfx" },
+    { "on",        NULL, 0, AV_OPT_TYPE_CONST, { .i64 = MFX_HYPERMODE_ON        },   INT_MIN, INT_MAX, VE, "dual_gfx" },
+    { "adaptive",  NULL, 0, AV_OPT_TYPE_CONST, { .i64 = MFX_HYPERMODE_ADAPTIVE  },   INT_MIN, INT_MAX, VE, "dual_gfx" },
+#endif
```

Comment:
Could you add the options when QSV_HAVE_HE is 0, like what we did in https://github.com/FFmpeg/FFmpeg/blob/master/libavfilter/vf_scale_qsv.c#L631-L635? If so, user needn't change the command line when using FFmpeg which is built against MFX_VERSION < 2.0

Reply:
I added this, thanks